File: Greedy/codeforces_1941B.py

# Problem link ===>> https://codeforces.com/problemset/problem/1941/B

def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        a = [int(x) for x in input().split()]
        for i in range(n - 2):
            if a[i] < 0:
                print('NO')
                return
            op = a[i]
            a[i] -= op
            a[i + 1] -= 2 * op
            a[i + 2] -= op
        if a[-1] != 0 or a[-2] != 0:
            print('NO')
        else:
            print('YES')


if __name__ == "__main__":
    main()

File: Greedy/codeforces_1941B.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1941/B
// submission link ===>> https://codeforces.com/contest/1941/submission/263027060

#include<bits/stdc++.h>
#define ll long long
using namespace std;

int main()
{
    ll t;
    cin >> t;

    for (ll i = 0; i < t; i++)
    {
        ll n;
        cin >> n;

        vector<ll> a(n);
        for (ll j = 0; j < n; j++)
        {
            cin >> a[j];
        }

        bool possible = true;
        for (ll j = 0; j < n - 2; j++)
        {
            if (a[j] < 0)
            {
                possible = false;
                break;
            }

            ll op = a[j];
            a[j] -= op;
            a[j + 1] -= 2 * op;
            a[j + 2] -= op;
        }

        if (possible && a[n - 1] == 0 && a[n - 2] == 0)
        {
            cout << "YES" << endl;
        }
        else
        {
            cout << "NO" << endl;
        }
    }

    return 0;
}
